src/main/scala/com/fulcrumgenomics/bam/FindTechnicalReads.scala

```diff
@@ -77,33 +93,66 @@ class FindTechnicalReads
       val r1 = iter.next()
       val r2 = if (iter.hasNext && iter.head.getReadName == r1.getReadName) Some(iter.next()) else None
       val recs = Seq(Some(r1), r2).flatten
-      if (recs.exists(r => matcher.matches(r.getReadBases))) {
-        recs.foreach(out.addAlignment)
-        n += recs.size
+
+      val isTechnical = if (isUnmappedTemplate(r1)) {
+        val hit      = recs.toStream.flatMap(r => matcher.findMatch(r.getReadBases)).headOption
```

Comment:
The match is tidier, you're right.  I wanted to keep the `Matcher` working in just sequence as I figure it might be more generally useful at some point, and it's only in the tagging that we currently care about the index.